### PR TITLE
Update FairScale on CI

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Related to #6152, #6937. As we move to FSDP we can deprecate the RPC sequential plugin.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by SeanNaren** [comment]
PyTorch 1.4 seems to be unsupported for FairScale which is fine; we just need to change some of the CI to not build fairscale for 1.4 and below + unavailable in imports

**Comment by carmocca** [comment]
Can you update https://github.com/PyTorchLightning/pytorch-lightning/blob/7c5ad1905d93515b90543b6855cc74e2c6bf6bf8/azure-pipelines.yml#L62 to not skip fairscale? assuming the docker fairscale is in sync with ours.

Also:

https://github.com/PyTorchLightning/pytorch-lightning/blob/aaccbeea2beb3b903428643d2ee520d2dd05b348/dockers/tpu-tests/Dockerfile#L38-L39

https://github.com/PyTorchLightning/pytorch-lightning/blob/aaccbeea2beb3b903428643d2ee520d2dd05b348/dockers/base-xla/Dockerfile#L103-L104

https://github.com/PyTorchLightning/pytorch-lightning/blob/23e8dff6b98f3429b31aa019bf126ad55d0f3ceb/.github/workflows/ci_test-full.yml#L83-L85

File: requirements/extra.txt [hunk 1]
```diff
@@ -8,5 +8,5 @@ torchtext>=0.5
 onnxruntime>=1.3.0
 hydra-core>=1.0
 # todo: when switch to standard package stream, drop `fairscale` from hard mocked docs libs
```
Comment (carmocca) [inline requirements/extra.txt:10]: @Borda resolve this TODO in this PR?

**Comment by SeanNaren** [comment]
FairScale has a dependency on the pytorch version being greater than 1.5, forcing the CI to install a later release: https://github.com/PyTorchLightning/pytorch-lightning/pull/7017/checks?check_run_id=2355656294#step:3:353 

this breaks the docker containers. This will require re-instating the need to remove FairScale from the extra requirements either in the CI or in the file, and only installing for CI testing. Not sure which is prefered.

**Comment by Borda** [comment]
That has to be updated in both, CI and dockers (as not all our CI uses dockers)

**Comment by SeanNaren** [comment]
> That has to be updated in both, CI and dockers (as not all our CI uses dockers)

I think this requires adding back some of the logic that @carmocca requested to remove, is this true?

**Comment by carmocca** [comment]
Just needs to install fairscale only if torch>=1.6.0 both in each Dockerfile and each config file

**Comment by SeanNaren** [comment]
After offline discussions we've decided to just fix the version of FairScale in the CI and remove it from the extras.txt due to the issue of having a user install the extras with a PyTorch version less than 1.6.

File: docs/source/conf.py [hunk 2]
```diff
@@ -335,7 +335,7 @@ def package_list_from_file(file):
 }
 MOCK_PACKAGES = []
 if SPHINX_MOCK_REQUIREMENTS:
-    MOCK_PACKAGES += ['fairscale']
+    MOCK_PACKAGES += []
```
Comment (carmocca) [inline docs/source/conf.py:338]:
```suggestion
```

**Comment by SeanNaren** [comment]
I'm a bit hesitant to have `pip install fairscale>=0.3.4` in our CI, should we fix this? i.e `pip install fairscale==0.3.4`? I don't want a breaking change to happen because of this version

**Comment by Borda** [comment]
> I'm a bit hesitant to have `pip install fairscale>=0.3.4` in our CI, should we fix this? i.e `pip install fairscale==0.3.4`? I don't want a breaking change to happen because of this version

that is hard to say as freeze will block/make a hard time for any who needs for other reason newer version... even the newer version will be fully compatible

**Comment by ananthsub** [comment]
@SeanNaren pinning to 0.3.4 would mean we don't have https://github.com/facebookresearch/fairscale/pull/620 pulled in

**Comment by Borda** [comment]
> @SeanNaren pinning to 0.3.4 would mean we don't have [facebookresearch/fairscale#620](https://github.com/facebookresearch/fairscale/pull/620) pulled in

correct so I would rather keep open interval and ban version if needed

**Comment by SeanNaren** [comment]
Ok I'll do a `>=` from sign now, and will monitor to ensure that if CI breaks we can ban versions

**Review by Borda — COMMENTED** [review]
this means we ll test only on GPU machine so let's add it also in the Azure config :]

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: .github/workflows/ci_test-full.yml [hunk 3]
```diff
@@ -83,10 +83,6 @@ jobs:
             req = open(fname).read().replace('>=', '==')
             open(fname, 'w').write(req)
```
Comment (carmocca) [inline .github/workflows/ci_test-full.yml:85]:
```suggestion
```

**Review by carmocca — APPROVED** [review]

**Pull request merged**